.github/workflows/npm.yml

```
@@ -0,0 +1,50 @@
+name: NPM
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "solidity/contracts/**"
+      - "solidity/package.json"
+      - "solidity/yarn.lock"
+  workflow_dispatch:
+
+jobs:
+  npm-compile-publish-contracts:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          registry-url: "https://registry.npmjs.org"
+          cache: "yarn"
+          cache-dependency-path: solidity/yarn.lock
+
+      - name: Resolve latest contracts
+        run: |
+          yarn add @keep-network/tbtc
```

We should use [`yarn upgrade`](https://classic.yarnpkg.com/en/docs/cli/upgrade/) as it will respect the semver ranges we configured in `package.json`. `yarn add` would install the latest package version.
```suggestion
          yarn upgrade @keep-network/tbtc
```